src/mongoose_account_api.erl

```diff
@@ -0,0 +1,320 @@
+    Result = ejabberd_auth:set_password(JID, Password),
+    format_change_password(Result).
+
+-spec check_account(jid:user(), jid:server()) -> check_account_result().
+check_account(User, Host) ->
+    JID = jid:make(User, Host, <<>>),
+    check_account(JID).
+
+-spec check_account(jid:jid()) -> check_account_result().
+check_account(JID) ->
+    case ejabberd_auth:does_user_exist(JID) of
+        true ->
+            {ok, io_lib:format("User ~s exists", [jid:to_binary(JID)])};
+        false ->
+            {user_does_not_exist, io_lib:format("User ~s does not exist", [jid:to_binary(JID)])}
+    end.
+
+-spec check_password(jid:user(), jid:server(), binary()) -> check_password_result().
+check_password(User, Host, Password) ->
+    JID = jid:make(User, Host, <<>>),
+    check_password(JID, Password).
+
+-spec check_password(jid:jid(), binary()) -> check_password_result().
+check_password(JID, Password) ->
+    case ejabberd_auth:does_user_exist(JID) of
+        true ->
+            case ejabberd_auth:check_password(JID, Password) of
+                true ->
+                    {ok, io_lib:format("Password '~s' for user ~s is correct",
+                                       [Password, jid:to_binary(JID)])};
+                false ->
+                    {incorrect, io_lib:format("Password '~s' for user ~s is incorrect",
+                                              [Password, jid:to_binary(JID)])}
+            end;
+        false ->
+            {user_does_not_exist,
+            io_lib:format("Password ~s for user ~s is incorrect because this user does not"
+                          " exist", [Password, jid:to_binary(JID)])}
+    end.
+
+-spec check_password_hash(jid:user(), jid:server(), string(), string()) ->
+    check_password_hash_result().
+check_password_hash(User, Host, PasswordHash, HashMethod) ->
+    JID = jid:make(User, Host, <<>>),
+    check_password_hash(JID, PasswordHash, HashMethod).
+
+-spec check_password_hash(jid:jid(), string(), string()) -> check_password_hash_result().
+check_password_hash(JID, PasswordHash, HashMethod) ->
+    AccountPass = ejabberd_auth:get_password_s(JID),
+    AccountPassHash = case HashMethod of
+        "md5" -> get_md5(AccountPass);
+        "sha" -> get_sha(AccountPass);
+        _ -> undefined
+    end,
+    case AccountPassHash of
+        undefined ->
+            {error, "Hash for password is undefined"};
+        PasswordHash ->
+            {ok, "Password hash is correct"};
+        _->
+            {incorrect, "Password hash is incorrect"}
+    end.
+
+-spec num_active_users(jid:server(), integer()) -> non_neg_integer().
+num_active_users(Domain, Days) ->
+    TimeStamp = erlang:system_time(second),
+    TS = TimeStamp - Days * 86400,
+    try
+        {ok, HostType} = mongoose_domain_api:get_domain_host_type(Domain),
+        mod_last:count_active_users(HostType, Domain, TS)
+    catch _:_ ->
+        0
+    end.
+
+-spec ban_account(jid:user(), jid:server(), binary() | string()) ->
```

Comment:
`binary() | string()` that spec is _evil_.

Should be just binary. also, any code that calls `ban_account` should provide a binary, of course.

Reply:
`kick_sessions` expects `binary` already, could you correct this?